sql/core/src/test/scala/org/apache/spark/sql/execution/SparkSqlSerializer2Suite.scala

```diff
@@ -57,6 +56,8 @@ class SparkSqlSerializer2DataTypeSuite extends SparkFunSuite {
   checkSupported(DecimalType(10, 5), isSupported = true)
   checkSupported(DecimalType.Unlimited, isSupported = true)
 
+  // If NullType is the only data type in the schema, we do not support it.
```

Comment:
oh, we have that test. The table `shuffle` used in `SparkSqlSerializer2Suite` contains all supported types and NullType.